unittests/test_edienergyscraper.py

```diff
@@ -434,3 +435,451 @@ async def test_mirroring(self, mocker, tmpdir_factory, datafiles, caplog):
+                    (EdifactFormatVersion.FV2310, None),
+                    (EdifactFormatVersion.FV2310, None),
+                    (EdifactFormatVersion.FV2310, None),
+                    (EdifactFormatVersion.FV2310, None),
+                    (EdifactFormatVersion.FV2310, None),
+                    (EdifactFormatVersion.FV2310, None),
+                    (EdifactFormatVersion.FV2310, None),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILMD]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILTS]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILTS]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILTS]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILTS]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILTS]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILTS]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILTS]),
+                    (EdifactFormatVersion.FV2310, [EdifactFormat.UTILTS]),
+                ],
+                id="All",
+            ),
+        ],
+    )
+    def test_get_edifact_format(
+        self, input_files: list[str], expected_results: list[(EdifactFormatVersion, EdifactFormat | None)]
+    ):
+        results = list()
+        ees = EdiEnergyScraper()
+        for file in input_files:
+            results.append(ees.get_edifact_format(Path(file)))
+
+        assert results == expected_results
```

Comment:
ich finde super, dass du alles abtestest, das gibt uns viel stabilität und coverage.

allerdings scheint mir die parametrisierung finde ich ein bisschen unintuitiv.
warum baust du im test eine liste auf statt die einzelnen werte einzeln abzutesten?

warum reicht es nicht, dass wir einen einzelnen filename als parameter des tests verwenden? und ein einzelnes tuple aus formatversion und format?

dann hast du nämlich statt 1 test mit 1 riesig großen parameter, 1 test mit vielen kleinen parametern. in der schreibweise wie es jetzt ist, muss ich ja z.b. den 47ten eintrag aus der ersten ewig langen liste mit dem 47ten eintrag aus der zweiten ewig langen liste vergleichen. das ist eher umständlich.

Reply:
Im Issue hieß es "Write a test for your function that is parametrized with all the filenames we currently encounter". Dass habe ich so verstanden als sollten wirklich alle Dateinamen getestet werden. Ich mache mir Gedanken, welchen bzw. welche 2, 3 Dateinamen am besten als Test verwendet werden.

Reply:
https://github.com/Hochfrequenz/edi_energy_scraper/pull/124/commits/f91af366cd4ec5f7c15e5f1bae17a80b9ff29707 <-- so hatte ich es mir vorgestellt